### PR TITLE
Remove machine config from operator

### PR DESCRIPTION
**Pull request by husky-parul**
This PR removes machine config dependencies from the operator

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,6 @@ require (
 	github.com/onsi/ginkgo/v2 v2.6.0
 	github.com/onsi/gomega v1.24.1
 	github.com/openshift/api v3.9.1-0.20190924102528-32369d4db2ad+incompatible
```
Comment (sthaha) [inline go.mod:9]: Do we need this now?

File: go.mod [hunk 2]
```diff
@@ -7,7 +7,6 @@ require (
...
-	github.com/openshift/machine-config-operator v0.0.1-0.20200913004441-7eba765c69c9
```
Comment (sthaha) [inline go.mod:10]: Could you please run `go.mod tidy`? 

It may be worth adding a lint for that WDTY?

File: controllers/kepler_controller.go [hunk 3]
```diff
@@ -88,7 +88,6 @@ func (r *KeplerReconciler) removePVC(logger logr.Logger, inst *keplerv1alpha1.Ke
 //+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=*,verbs=*
 //+kubebuilder:rbac:groups=core,resources=nodes/metrics;nodes/proxy;nodes/stats,verbs=get;list;watch
 //+kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,verbs=get;list;watch;create;update;patch;delete;use
-//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigs;machineconfigpools,verbs=get;list;watch;create;update;patch;delete;use
```
Comment (sthaha) [inline controllers/kepler_controller.go:91]: Shouldn't we `make bundle` and update the bundle as well?

**Review by sthaha — COMMENTED** [review]
I tested this running the operator out of cluster against openshift 4.13-nightly and querying kepler exporter service for metrics which seems to work fine. 

I think we should ensure
-  the bundle is up to date 
-  go mod tidy is done as well 

On an different topic, why are we vendoring (go mod vendor) instead of relying on `go mod` itself to resolve the dependencies ?

**Comment by husky-parul** [comment]
> the bundle is up to date

We already have the steps to make/validate and test bundle in the Makefile. I want to automate the bundle part as a GHA post PR merge into this [repo](https://github.com/sustainable-computing-io/community-operators-prod) through which we raise PR in the community catalog. Maybe we can open an issue and discuss this pipeline? 


> go mod tidy is done as well

Done

> On an different topic, why are we vendoring (go mod vendor) instead of relying on go mod itself to resolve the dependencies ?

I will update the `.gitignore` for `vendor`

**Comment by sthaha** [comment]
/lgtm , please feel free to merge @husky-parul 

@vprashar2929, just FYI something to test on Openshift :)

**Pull request merged**